zaza/openstack/charm_tests/ovn/setup.py

```diff
@@ -0,0 +1,151 @@
+
+import zaza
+
+import zaza.openstack.charm_tests.test_utils as test_utils
+
+
+class _OVNSetupHelper(test_utils.BaseCharmTest):
+    """Helper class to get at the common `config_change` helper."""
+
+    @staticmethod
+    def _get_instance_mtu_from_global_physnet_mtu():
+        """Calculate instance mtu from Neutron API global-physnet-mtu.
+
+        :returns: Value for instance mtu after migration.
+        :rtype: int
+        """
+        n_api_config = zaza.model.get_application_config('neutron-api')
+
+        # NOTE: we would have to adjust this calculation if we use IPv6 tunnel
+        # endpoints
+        GENEVE_ENCAP_OVERHEAD = 38
+        IP4_HEADER_SIZE = 20
+        return int(n_api_config['global-physnet-mtu']['value']) - (
+            GENEVE_ENCAP_OVERHEAD + IP4_HEADER_SIZE)
+
+    def _configure_apps(self, apps, cfg,
+                        first_match_raise_if_none_found=False):
+        """Conditionally configure a set of applications.
+
+        :param apps: Applications.
+        :type apps: Iterator[str]
+        :param cfg: Configuration to apply.
+        :type cfg: Dict[str,any]
+        :param first_match_raise_if_none_found: When set the method will
+                                                configure the first application
+                                                it finds in the model and raise
+                                                an exception if none are found.
+        :type first_match_raise_if_none_found: bool
+        :raises: RuntimeError
+        """
+        for app in apps:
+            try:
+                zaza.model.get_application(app)
+                for k, v in cfg.items():
+                    logging.info('Setting `{}` to "{}" on "{}"...'
+                                 .format(k, v, app))
+                    with self.config_change(cfg, cfg, app):
+                        # The intent here is to change the config and not
+                        # restore it. We accomplish that by passing in the same
+                        # value for default and alternate.
+                        #
+                        # The reason for using the `config_change` helper for
+                        # this is that it already deals with all the
+                        # permutations of config already being set etc and does
+                        # not get into trouble if the test bundle already has
+                        # the values we try to set.
+                        if first_match_raise_if_none_found:
+                            break
+                        else:
+                            continue
+                else:
+                    if first_match_raise_if_none_found:
+                        raise RuntimeError(
+                            'None of the expected apps ({}) are present in '
+                            'the model.'
+                            .format(apps)
+                        )
+            except KeyError:
+                pass
```

Comment:
This caused a regression, putting back to draft for now

Reply:
This actually worked, it was me forgetting to put the configure job into tests.yaml for the neutron-gateway charm ;)